src/Migrations.php

```diff
@@ -288,6 +288,20 @@ public function getManager($config = null)
 
             $this->manager = new CakeManager($config, $this->output);
         } elseif ($config !== null) {
+            $defaultEnvironment = $config->getEnvironment('default');
+            try {
+                $environment = $this->manager->getEnvironment('default');
+                $oldConfig = $environment->getOptions();
+                unset($oldConfig['connection']);
+                if ($oldConfig == $defaultEnvironment) {
```

Comment:
I didn't say order, i meant checking with `===` for consistency as it's used everywhere in the code when doing checks like that. Sorry for not being clear.